src/LazyOperations/Bloating.jl

```diff
@@ -0,0 +1,135 @@
+export Bloating
+
+"""
+    Bloating{N<:AbstractFloat, S<:LazySet{N}} <:LazySet{N}
+
+Type that represents a (possibly negative) uniform expansion of a convex set in
+a given norm (also known as *bloating*).
+
+### Fields
+
+- `X` -- convex set
+- `ε` -- (positive) bloating factor
+- `p` -- ``p``-norm (``≥ 1``)
+"""
+struct Bloating{N<:AbstractFloat, S<:LazySet{N}} <:LazySet{N}
+    X::S
+    ε::N
+    p::N
+
+    function Bloating(X::S, ε::N, p::N=N(2)) where {N<:AbstractFloat,
+                                                    S<:LazySet{N}}
+        @assert ε > zero(N) "bloating requires a distance > 0, but $ε was given"
+        @assert p >= one(N) "bloating requires a norm >= 1, but $p was given"
+
+        return new{N, S}(X, ε, p)
+    end
+end
+
+"""
+    dim(B::Bloating)
+
+Return the dimension of a bloated convex set.
+
+### Input
+
+- `B` -- bloated convex set
+
+### Output
+
+The ambient dimension of the bloated set.
+"""
+function dim(B::Bloating)
+    return dim(B.X)
+end
+
+"""
+    σ(d::AbstractVector{N}, B::Bloating{N}) where {N<:AbstractFloat}
+
+Return the support vector of a bloated convex set in a given direction.
+
+### Input
+
+- `d` -- direction
+- `B` -- bloated convex set
+
+### Output
+
+The support vector of the bloated set in the given direction.
+"""
+function σ(d::AbstractVector{N}, B::Bloating{N}) where {N<:AbstractFloat}
+    return σ(d, B.X) + normalize(d, B.p) * B.ε
```

Comment:
here i would assert that the norm of d is not zero, to avoid NaNs in the output.

Reply:
:thinking: : the formula doesn't give what i expect for the second term:

```julia
d = rand(3)
p = 2.5
ε = 1e-2

normalize(d, p) * ε
3-element Array{Float64,1}:
 0.0034857477766639185
 0.009630890087713    
 0.002005076494778946

σ(d, Ballp(p, zeros(3), ε))
3-element Array{Float64,1}:
 0.004634748147714765 
 0.009125845625116528 
 0.0032056502022087315
```

Reply:
You are right, my formula was wrong. Now I do it the lazy way and actually compute the result for the ball.